Source/inv.cpp

```diff
@@ -347,50 +537,28 @@ void CheckInvPaste(Player &player, Point cursorPosition)
 	if (slot >= SLOTXY_BELT_FIRST && slot <= SLOTXY_BELT_LAST)
 		il = ILOC_BELT;
 
-	item_equip_type desiredIl = player.GetItemLocation(player.HoldItem);
-	if (il == ILOC_ONEHAND && desiredIl == ILOC_TWOHAND)
-		il = ILOC_TWOHAND;
-
-	int8_t it = 0;
-	if (il == ILOC_UNEQUIPABLE) {
-		int ii = slot - SLOTXY_INV_FIRST;
-		if (player.HoldItem._itype == ItemType::Gold) {
-			if (player.InvGrid[ii] != 0) {
-				int8_t iv = player.InvGrid[ii];
-				if (iv > 0) {
-					if (player.InvList[iv - 1]._itype != ItemType::Gold) {
-						it = iv;
-					}
-				} else {
-					it = -iv;
-				}
-			}
-		} else {
-			// check that the item we're pasting only overlaps one other item (or is going into empty space)
-			unsigned originCell = static_cast<unsigned>(slot - SLOTXY_INV_FIRST);
-			for (unsigned rowOffset = 0; rowOffset < static_cast<unsigned>(itemSize.height * InventorySizeInSlots.width); rowOffset += InventorySizeInSlots.width) {
-				for (unsigned columnOffset = 0; columnOffset < static_cast<unsigned>(itemSize.width); columnOffset++) {
-					unsigned testCell = originCell + rowOffset + columnOffset;
-					// FindTargetSlotUnderItemCursor returns the top left slot of the inventory region that fits the item, we can be confident this calculation is not going to read out of range.
-					assert(testCell < sizeof(player.InvGrid));
-					if (player.InvGrid[testCell] != 0) {
-						int8_t iv = std::abs(player.InvGrid[testCell]);
-						if (it != 0) {
-							if (it != iv) {
-								// Found two different items that would be displaced by the held item, can't paste the item here.
-								return;
-							}
-						} else {
-							it = iv;
-						}
-					}
-				}
-			}
-		}
-	} else if (il == ILOC_BELT) {
+	if (il == ILOC_ONEHAND && desired_loc == ILOC_TWOHAND)
+		return ILOC_TWOHAND;
+
+	return il;
+}
+
+void CheckInvPaste(Player &player, Point cursorPosition)
+{
+	const Size itemSize = GetInventorySize(player.HoldItem);
+
+	const int slot = FindTargetSlotUnderItemCursor(cursorPosition, itemSize);
+	if (slot == NUM_XY_SLOTS)
+		return;
+
+	auto const desired_loc = player.GetItemLocation(player.HoldItem);
+	const auto il = GetItemEquipType(slot, desired_loc, player);
+	if (il == ILOC_BELT) {
 		if (!CanBePlacedOnBelt(player, player.HoldItem))
 			return;
-	} else if (desiredIl != il) {
+	}
+
+	if ((il != ILOC_UNEQUIPABLE) && (desired_loc != il)) {

```

Comment:
Previously there was and else in between these, with out it it is now not possible to place items on the belt as `desired_loc` is never `ILOC_BELT`.
```suggestion
	} else if ((il != ILOC_UNEQUIPABLE) && (desired_loc != il)) {
```